framework/web/Response.php

```diff
@@ -364,16 +364,10 @@ protected function sendCookies()
             return;
         }
         $request = Yii::$app->getRequest();
-        if ($request->enableCookieValidation) {
-            if ($request->cookieValidationKey == '') {
-                throw new InvalidConfigException(get_class($request) . '::cookieValidationKey must be configured with a secret key.');
-            }
-            $validationKey = $request->cookieValidationKey;
-        }
         foreach ($this->getCookies() as $cookie) {
             $value = $cookie->value;
-            if ($cookie->expire != 1  && isset($validationKey)) {
-                $value = Yii::$app->getSecurity()->hashData(serialize([$cookie->name, $value]), $validationKey);
+            if ($cookie->expire != 1 && $request->enableCookieValidation && !in_array((string)$cookie->name, $request->cookieValidationWhitelist, true)) {
```

Comment:
What is the use of strict check combined with a manual type cast?
This is effecitvely the same as using a non-strict check.